gdx/src/com/badlogic/gdx/graphics/g3d/decals/AlphaTestGroupStrategy.java

```diff
@@ -0,0 +1,251 @@
+		  shader.setUniformMatrix("u_projectionViewMatrix", camera.combined);
+		  shader.setUniformi("u_texture", 0);
+	 }
+
+	 @Override public void afterGroups () {
+		  Gdx.gl.glDisable(GL20.GL_DEPTH_TEST);
+	 }
+
+	 /**
+	  * Creates default shader.
+	  *
+	  * @param bayerFilter 2^maxBAYERVALUE is the width and height of the Bayer filter.
+	  *                    As 16x16 holds all possible 255 alpha values, 4 is the functional
+	  *                    maximum size.
+	  */
+	 private void createDefaultShader (final BayerFilter bayerFilter) {
+		  String vertexShader = "attribute vec4 " + ShaderProgram.POSITION_ATTRIBUTE + ";\n" //
+			  + "attribute vec4 " + ShaderProgram.COLOR_ATTRIBUTE + ";\n" //
+			  + "attribute vec2 " + ShaderProgram.TEXCOORD_ATTRIBUTE + "0;\n" //
+			  + "uniform mat4 u_projectionViewMatrix;\n" //
+			  + "varying vec4 v_color;\n" //
+			  + "varying vec2 v_texCoords;\n" //
+			  + "\n" //
+			  + "void main()\n" //
+			  + "{\n" //
+			  + "   v_color = " + ShaderProgram.COLOR_ATTRIBUTE + ";\n" //
+			  + "   v_texCoords = " + ShaderProgram.TEXCOORD_ATTRIBUTE + "0;\n" //
+			  + "   gl_Position =  u_projectionViewMatrix * " + ShaderProgram.POSITION_ATTRIBUTE + ";" + "\n" //
+			  + "}\n";
+
+		  String dfragmentShader = "#ifdef GL_ES\n" //
+			  + "precision mediump float;\n" //
+			  + "#endif\n" //
+			  + "\n" //
+			  + "varying vec4 v_color;\n" //
+			  + "varying vec2 v_texCoords;\n" //
+			  + "uniform sampler2D u_texture;\n" //
+			  + "\n" //
+			  + "float GetBayer()\n" //
+			  + "{\n" //
+			  + "float finalBayer   = 0.0;\n" //
+			  + "float finalDivisor = 0.0;\n" //
+			  + "float layerMult\t   = 1.0;\n" //
+			  + "\n" //
+			  + "for(int bayerLevel = " + bayerFilter.size + "; bayerLevel >= 1; bayerLevel--)\n" //
+			  + "{\n" //
+			  + "float bayerSize \t= exp2( float(bayerLevel)) * 0.5;\n" //
+			  + "vec2 bayercoord \t= mod( floor(gl_FragCoord.xy / bayerSize) , 2.0);\n" //
+			  + "layerMult \t\t   *= 4.0;\n" //
+			  + "\n" //
+			  + "float line0202 = bayercoord.x * 2.0;\n" //
+			  + "\n" //
+			  + "finalBayer += mix(line0202,3.0 - line0202,bayercoord.y) / 3.0 * layerMult;\n" //
+			  + "finalDivisor += layerMult;\n" //
+			  + "}\n" //
+			  + "return finalBayer / finalDivisor;\n" //
+			  + "}\n" //
+			  + "\n" //
+			  + "void main()\n" //
+			  + "{\n" //
+			  + "float stipple =GetBayer(); \n" //
+			  + "vec4 tex2D =  texture2D(u_texture, v_texCoords);\n" //
+			  + "vec4 tinted = v_color * tex2D; \n" //
+			  + "if( tinted.a <= stipple ) { discard; }\n" //
+			  + "else\n" //
+			  + "{ gl_FragColor = tinted; }\n" //
+			  + "}"; //
+
+		  //Tetting Bayer shader
+		  String fragmentShader = "#ifdef GL_ES\n" //
+			  + "precision mediump float;\n" //
+			  + "#endif \n" //
+			  + "varying vec4 v_color;\n" //
+			  + "varying vec2 v_texCoords;\n" //
+			  + "uniform sampler2D u_texture;\n" //
+			  + "\n" //
+			  + "#define MAX_LEVEL 4\n" //
+			  + "float GetBayer() {\n" //
```

Comment:
why calculating for every fragment when we could use a lookup table (like in your original PR)?

Reply:
this one works on webgl and the lookup one can't unfortunately. variable access to arrays not allowed and for four iterations wtih this you get the equivalent of 16x16 dither block, although a single pass looks ok imo